rotkehlchen/chain/evm/decoding/decoder.py

```diff
@@ -437,13 +444,17 @@ def get_and_decode_undecoded_transactions(
                 limit=limit,
                 addresses=addresses,
             )
-            self.decode_transaction_hashes(ignore_cache=False, tx_hashes=hashes)
+            _, refresh_balances = self.decode_transaction_hashes(
+                ignore_cache=False,
+                tx_hashes=hashes,
+            )
+            self.post_process(refresh_balances=refresh_balances)
```

Comment:
If I read correctly the code the core of the decoding process always happens in `decode_transaction_hashes` (this function in the case of the task manager `get_and_decode_undecoded_transactions` and `decode_transaction_hashes` for `rest.py`.

So instead of further propagating the `refresh_balances` why not set it to `False` (you already do this) at the beggining of `decode_transaction_hashes` and before exiting `decode_transaction_hashes` you just execute

```
self.post_process(refresh_balances=refresh_balances)
```

Why this:

1. the decoding logic and what is needed to be executed is contained in the decoder class. Right now the `rest.py` file is required to execute logic that it shouldn't know about
2. You don't propagate values outside of where they are needed

Reply:
Yes, probably it makes sense. I agree that the way it is currently made is bad.

The reason why I did it the way I did it is because things like `decode_transaction` are also called separately, not only in `decode_transaction_hashes`, so was a bit tricky there to automatically handle post processing. But I think I have an idea about it.